src/python/WMCore/ReqMgr/Service/Request.py

```diff
@@ -74,25 +71,20 @@ def _validateRequestBase(self, param, safe, valFunc, requestName=None):
         data = cherrypy.request.body.read()
         if data:
             request_args = json.loads(data)
-            if requestName:
-                request_args["RequestName"] = requestName
-
         else:
-            # actually this is error case
-            # cherrypy.log(str(param.kwargs))
             request_args = {}
-            for prop in param.kwargs:
+        # In case key args are also passed and request body also exists.
+        # If the request.body is dictionally update the key args value as well
+        if isinstance(request_args, dict):
+            for prop in param.kwargs.keys():
                 request_args[prop] = param.kwargs[prop]
-
-            for prop in request_args:
                 del param.kwargs[prop]
```

Comment:
My suggestion was to use the `pop` dict method to get+remove that key/value. Two lines become a single one :)